### PR TITLE
feat: enable nativeUserMode on OpenShift with devworkspace by default

### PR DESCRIPTION
**Pull request by sparkoo**
Signed-off-by: Michal Vala <mvala@redhat.com>

### What does this PR do?
When che-operator is deploying on OpenShift with DevWorkspaces, enable `nativeUserMode` by default.

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20203

### How to test this PR?

1. `deploy with chectl server:deploy --platform=openshift --installer=operator --workspace-engine=dev-workspace --che-operator-image=quay.io/mvala/che-operator:gh20203-nativeUserModeEnable` (chectl will get stuck on Keycloak, but Che will deploy fine)
2. `nativeUserMode` should be enabled by default

### PR Checklist
          
[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-nightly-olm-bundle)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: controllers/che/checluster_controller.go [hunk 1]
```diff
@@ -315,6 +315,16 @@ func (r *CheClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		}
 	}
 
+	if util.IsOpenShift && instance.Spec.DevWorkspace.Enable && instance.Spec.Auth.NativeUserMode == nil {
+		newNativeUserModeValue := util.NewBoolPointer(true)
+		if !reflect.DeepEqual(newNativeUserModeValue, instance.Spec.Auth.NativeUserMode) {
```
Comment (tolusha) [inline controllers/che/checluster_controller.go:320]: This check is redundant.

**Review by tolusha — APPROVED** [review]
Pls rebase on master and let's wait for a green light.

File: controllers/che/checluster_controller.go (hunk 1 above)
Comment (sparkoo) [reply controllers/che/checluster_controller.go]: yes, removed

**Comment by tolusha** [comment]
`ci/prow/v7-stable-to-nightly`  is expected to fail (https://github.com/eclipse/che/issues/20235)

**Comment by sparkoo** [comment]
devworkspace might be failing because of https://github.com/che-incubator/chectl/pull/1502

**Comment by tolusha** [comment]
It means we need chectl PR merged first

**Comment by sparkoo** [comment]
> It means we need chectl PR merged first

I've updated the chectl PR https://github.com/che-incubator/chectl/pull/1502

**Review by tolusha — APPROVED** [review]

**Pull request merged**